### ISSUE TITLE
emerge: there are no ebuilds to satisfy "=dev-lang/dart-3*".

### ISSUE THREAD

**Issue by bugalo** [jgenco/jgenco-overlay#34]
"=dev-lang/dart" doesn't exist.

It can be fixed by removing the "BDEPEND" on "=dev-lang/dart-3*" in "dart-sass/dart-sass-1.70.0.ebuild". It will still pull "=dev-lang/dart-bin-3*", which exists.

**Comment by jgenco** [comment]
**First**: Is portage trying to pull in `dart` and not `dart-bin`?  It should just pull  in `dev-lang/dart-bin` , because of :
https://github.com/jgenco/jgenco-overlay/blob/79d945b4cb61b8965ff815932785348797b6be2d/dev-lang/dart-sass/dart-sass-1.70.0.ebuild#L156-L164
My understanding it should try to satisfy `=dev-lang/dart-3*` and when that fails use `=dev-lang/dart-bin-3*`.

**History of why `dev-lang/dart` doesn't exist**:
When I was first putting quarto's dependencies together. I had trouble finding source files for dart.  I guess your suppose to git pull a giant repo or two and than build dart. An ancient [ebuild](https://gist.github.com/nidev/65c142dd82b2d9f09e5d) example of this.  I did find an [ebuild](https://github.com/mausys/portage/blob/master/dev-lang/dart/dart-2.12.2.ebuild) (and a slightly newer [version](https://github.com/funtoo/kit-fixups/blob/master/lang-kit/curated/dev-lang/dart/dart-2.3.2.ebuild) ) that used a tarball instead of using VCS.  The problem came when I was looking at  [dart-2.12.2-python3.patch](https://github.com/mausys/portage/blob/master/dev-lang/dart/files/dart-2.12.2-python3.patch). It stated "`# Allow 8Gb per link on Linux because Gold is quite memory hungry`".  At the time I _only_ had 8G of memory so I decided to just write `dev-lang/dart-bin`.

TLDR: Writing it was hard / impossible at the time so I just made the bin version.

 I put the non _bin_ dart there so if ::gentoo someday or another overlay had a `dev-lang/dart` the user could use that if they wanted.

**Comment by bugalo** [comment]
Thank you for the explanation. I am not an expert, but I understand, as you do, that portage should skip "=dev-lang/dart-3*" and pull "=dev-lang/dart-bin-3*". However, it was failing for me. Maybe there is something wrong on my end?

**Comment by jgenco** [comment]
With dart-bin not emerged ... presumably your current status.

I was able to get your error by adding:
`dev-lang/dart-bin` or `*/*-bin` to my [packatge.mask](https://wiki.gentoo.org/wiki//etc/portage/package.mask)  or
`dev-lang/dart-bin -amd64` or `*/*-bin -amd64` to my [package.accept_keyword](https://wiki.gentoo.org/wiki//etc/portage/package.accept_keyword) file.

I could see a Gentoo user adding `*/*-bin` to their package.mask file(s).

You could test to see if this or some other problem  by running `emerge -p dev-lang/dart-bin` and see what that error tells you.

**Comment by bugalo** [comment]
I think I understand now what is happening. When I add an overlay, I mask all of its packages in portage.mask, in this case with "*/*::jgenco". Then I unmask the required packages one by one when needed. In this case, "dev-lang/dart-bin" was masked, so emerge was trying to pull "dev-lang/dart" and complained that it doesn't exist. I cannot test it at the moment, because I have "dev-lang/dart-bin" already installed. I'll try to reproduce it later and let you know.

Thanks you.

**Comment by jgenco** [comment]
I added package `dev-lang/dart` with commit c40ec345ef9c88df240cd1eb667d9f306e323e0c.  It probably at alpha stage right now.  The next time dart-sass needs updated or llvm-18 gets old (19 doesn't work), I'll probably work on it some more then.  
Currently the `LICENSE` is incomplete but I assume it has similar licenses as other v8 base programs.
If any one wants to use `dev-lang/dart` you need to add `dev-lang/dart **` to `/etc/portage/packages.accept_keywords`
